src/solver/infeasible-problem-analysis/problem.h

```diff
@@ -11,22 +11,77 @@ namespace Antares
 {
 namespace Optimization
 {
-class InfeasibleProblemAnalysis
+
+
+class SingleAnalysis
 {
 public:
-    InfeasibleProblemAnalysis() = delete;
-    explicit InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx);
-    InfeasibleProblemReport produceReport();
+    SingleAnalysis(std::shared_ptr<operations_research::MPSolver> problem_);
+    virtual void run() = 0;
+    virtual void printReport() = 0;
+    virtual std::string title() = 0;
+    bool hasDetectedInfeasibilityCause() { return hasDetectedInfeasibilityCause_; }
+
+protected:
+    std::shared_ptr<operations_research::MPSolver> problem_;
+    bool hasDetectedInfeasibilityCause_ = false;
+};
+
+
+class SlackVariablesAnalysis : public SingleAnalysis
+{
+    using SingleAnalysis::SingleAnalysis;
+public:
+    void run() override;
+    void printReport() override;
+    std::string title() override { return "Slack variables analysis"; }
 
 private:
     void buildObjective() const;
     void addSlackVariables();
     operations_research::MPSolver::ResultStatus Solve() const;
 
-    std::unique_ptr<operations_research::MPSolver> mSolver;
-    std::vector<const operations_research::MPVariable*> mSlackVariables;
+    std::vector<const operations_research::MPVariable*> slackVariables_;
     const std::string constraint_name_pattern = "^AreaHydroLevel::|::hourly::|::daily::|::weekly::|^FictiveLoads::";
-                                    
+};
+
+struct VariableBounds
+{
+    VariableBounds(std::string var_name, double low_bound, double up_bound) 
+        : name(var_name), lowBound(low_bound), upBound(up_bound)
+    {}
+
+    std::string name;
+    double lowBound;
+    double upBound;
+};
+
+class VariablesBoundsAnalysis : public SingleAnalysis
+{
+    using SingleAnalysis::SingleAnalysis;
+public:
+    void run() override;
+    void printReport() override;
+    std::string title() override { return "Variables bounds analysis"; }
+
+private:
+    void storeIncorrectVariable(std::string name, double lowBound, double upBound);
+    bool foundIncorrectVariables();
+
+    std::vector<VariableBounds> incorrectVars_;
+};
+
+class InfeasibleProblemAnalysis
+{
+public:
+    InfeasibleProblemAnalysis() = delete;
+    explicit InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx);
+    void run();
+    void printReport();
```

Comment:
Would be better to return a report object from the `run` method, to make the workflow more natural and separate roles:
```
auto report = analysis.run();
report.print();
```

Reply:
- About the **problem** as an argument of the **run(...)** method  : 
  In the current state of the code, we have 3 concrete classes : an analyzer (which supervises the whole analysis), and the 2 different analyses. Each analysis owns the problem, but the analyzer does not see the problem at all : it just knows if an analysis failed to find the reason of unfeasibility or succeeded, and if it must stop the analyse there or run another one. 
Should the analyzer see the problem ? At least it's a dependency it does not have.
So in this context, passing the problem to an analysis run function is impossible.
- About the **returned report** :  
  Currently the analyzer has the given interface  : **void run()** and **void printReport()**. So a client of the analyser needs to call the 2 public methods. Splitting the the whole analyze in 2 parts can make the code both clear (not too many calls) and informative (we know more precisely what the analyzer do). This can be discussed though.
But if we choose the leave the analyzer as it is, it may be neither easy nor consistent that the run method returns a report.